.github/workflows/main.yml

```diff
@@ -0,0 +1,24 @@
+name: Python Test
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  python-script-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout project sources
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # so that spotless can ratchet; see https://github.com/diffplug/spotless/issues/710
```

Comment:
This comment, at least, is not accurate (we're not using spotless, which is a Java code formatting tool). I don't object to `fetch-depth: 0` in principle, but we should remove this comment.